LiteCore/Query/QueryParser.cc

```diff
@@ -1132,6 +1164,102 @@ namespace litecore {
         }
     }
 
+    namespace {
+        const char* const metaKeys[][2] = {
+            {"id",         "f1rK.b3xI2bA6" "id"},
+            {"sequence",   "f1rK.b3xI2bA6" "sequence"},
+            {"deleted",    "f1rK.b3xI2bA6" "deleted"},
+            {"expiration", "f1rK.b3xI2bA6" "expiration"},
+            {"revisionID", "f1rK.b3xI2bA6" "revisionID"}
+        };
+        const unsigned kMetaKeysCount = sizeof(metaKeys) / sizeof(metaKeys[0]);
+        const char* const kMetaCryptic = "f1rK.b3xI2bA6";
+
+        Retained<MutableDict> buildMetaDict(const string& dbAlias) {
+            static Retained<MutableDict> ret;
+            if (!ret) {
```

Comment:
This is not thread-safe, especially since each caller to this method mutates the contents of the static dict. Why does it need to be static at all?

Reply:
Oh, just wanted to save it for the sake of consistency. If threading is a concern, then, I cannot keep it. I am clear of the threading model here, but with concurrent threads, other parts would fail, such as _sql.